sdk/eventhubs/azure-eventhubs/src/samples/java/com/azure/messaging/eventhubs/eventprocessor/ConsolePartitionProcessor.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs.eventprocessor;
+
+import com.azure.messaging.eventhubs.CheckpointManager;
+import com.azure.messaging.eventhubs.EventData;
+import com.azure.messaging.eventhubs.eventprocessor.models.PartitionContext;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A simple implementation of a partition processor that logs the methods called in this implementation to a console.
+ */
+public class ConsolePartitionProcessor implements Subscriber<EventData> {
+
+    private final Logger logger = LoggerFactory.getLogger(ConsolePartitionProcessor.class);
+    private final PartitionContext partitionContext;
+    private final CheckpointManager checkpointManager;
+    private Subscription subscription;
+
+    public ConsolePartitionProcessor(PartitionContext partitionContext,
+        CheckpointManager checkpointManager) {
+        this.partitionContext = partitionContext;
+        this.checkpointManager = checkpointManager;
+        logger.info("Initializing partition processor: event hub name = " + partitionContext
```

Comment:
Even though this is a sample, we should be showing best practises by using parameterised logging to defer calculation of the parameters if the log level is not set.
https://www.slf4j.org/faq.html#logging_performance